Updating Design and Adding Code of Conduct

JamesDearlove opened the pull request:
I've gone ahead and updated the design for the Slack inviter to take advantage of the static UQCS Bulma and looks similar to the updated membership form.

While I was at it, I've also added a disclaimer about our code of conduct.

![image](https://user-images.githubusercontent.com/39483549/106292557-afdd5080-6298-11eb-95c8-2084ccb5463c.png)

JamesDearlove committed:
Updating Slack site to use bulma

nicklambourne reviewed (changes requested):
Would it be worth taking advantage of the reCaptcha dark theme here and wherever else it's used?
https://wpforms.com/developers/how-to-change-the-captcha-theme-on-google-checkbox-v2-recaptcha/

I think it would also be worth having the Code of Conduct be attached to a checkbox that enables the submission button. People still may not read it, but at least they'll know they're agreeing to some set of terms.

nicklambourne commented:
Also, I know @mattchrlw has an unhealthy obsession with left justification, but I'd seriously consider switching everything (except input labels) back to being centred.

mattchrlw commented:
It looks so good left-justified though 🥺 but it might need a bit more inter-element padding to keep it clean looking.

katrinafyi commented on public/css/style.css:
```diff
@@ -1,188 +1,27 @@
-@import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;700;900&display=swap');
```
Hooray for having a unified style sheet 🎉

katrinafyi commented on views/error.pug:
```diff
@@ -11,13 +11,15 @@ html
     meta(name="viewport", content="width=device-width, initial-scale=1")
     title #{__('TITLE', community)}
+    link(href="//static.uqcs.org/bulma/main.css" rel="stylesheet" type="text/css")
     link(href="css/style.css", rel="stylesheet", type="text/css")
```
Would this 404 since you deleted the file?

JamesDearlove replied:
The style file still exists as it contains some site specific CSS

katrinafyi commented on views/index.pug:
```diff
@@ -11,30 +11,38 @@ html
     meta(name="viewport", content="width=device-width, initial-scale=1")
     title #{__('TITLE', community)}
+    link(href="//static.uqcs.org/bulma/main.css" rel="stylesheet" type="text/css")
     link(href="css/style.css", rel="stylesheet", type="text/css")
```
Same problem with deleted file.

katrinafyi commented on views/index.pug:
```diff
@@ -11,30 +11,38 @@ html
+          .column#body
+            .my-1#header
+              img(src="images/logo.svg", onerror="this.src='images/logo.svg.png'", id="logo")
+              h4(class="title is-4 my-5") #{__('HEADER', community)}
```
Title doesn't need to mention UQCS or email since our full logo is directly above it and email is below. (I think this text is set elsewhere.)

katrinafyi commented on views/index.pug:
```diff
@@ -11,30 +11,38 @@ html
+              
+              .buttons.wide.mb-1
+                input(class="button is-link" type="submit", value="Join").submit
+            p(class="help") By joining the UQCS Slack, you agree to have read and understood the 
```
This doesn't actually require they agree to abide by the CoC 😛

katrinafyi commented on views/index.pug:
```diff
@@ -11,30 +11,38 @@ html
+              .buttons.wide.mb-1
+                input(class="button is-link" type="submit", value="Join").submit
+            p(class="help") By joining the UQCS Slack, you agree to have read and understood the 
+              a(href="https://uqcs.org/code-of-conduct" target="_blank") Code of Conduct
```
Nit: full stop.

katrinafyi commented on views/index.pug:
```diff
@@ -11,30 +11,38 @@ html
+            p(class="help") By joining the UQCS Slack, you agree to have read and understood the 
+              a(href="https://uqcs.org/code-of-conduct" target="_blank") Code of Conduct
+            p(class="help") Already a slack user? 
               a(href=`https://${slackUrl}`) Sign in to #{community} here
```
Personally, I would like to see this be a button or at least alongside the Join button. Bulma's `.buttons` class could help: https://bulma.io/documentation/elements/button/

JamesDearlove replied:
IMO, having a button to sign in that is next to the join button in a UX sense, does not make sense here. As this is a form, by moving the login button next to the join button you imply that the user must fill out the form in order to sign into the Slack which is not the case.

When separating it as a link at the bottom, you imply to the user that to login to the Slack you need to navigate to a different website.

katrinafyi replied:
I think it makes sense. Sign up forms commonly offer a log in button (e.g. Google login) alongside the next button. My problem with the link is that it's very hard to notice, especially now there's the CoC link as well. 

Ideally, we should make it more obvious that returning users can go straight to https://uqcs.slack.com. Even if it's not a dedicated button, the text could be bigger (e.g. not .help and just normal size).

katrinafyi commented on views/result.pug:
```diff
@@ -11,11 +11,13 @@ html
     meta(name="viewport", content="width=device-width, initial-scale=1")
     title #{__('TITLE', community)}
+    link(href="//static.uqcs.org/bulma/main.css" rel="stylesheet" type="text/css")
     link(href="css/style.css", rel="stylesheet", type="text/css")
```
See above re deleted file.

katrinafyi commented on views/index.pug:
```diff
@@ -11,30 +11,38 @@ html
-            p Already a slack user? 
+                .field
+                  .control
+                    div(class="g-recaptcha", data-sitekey=recaptchaSiteKey)
```
To avoid the inconsistent colour and width of the recaptcha, would it be possible to use the [invisible recaptcha](https://developers.google.com/recaptcha/docs/invisible)? It would work in the background and only present a challenge on clicking.

katrinafyi reviewed (changes requested):
Screenshots look good. All minor comments.

Agree with checkbox for CoC.  Unfortunately, I have inherited Matt's obsession for left-justification. You could tweak the column and button widths to get it looking better, I think.

Also, _super nit_: commit title should be more descriptive.

katrinafyi commented:
As a note for later, I think this could be easily integrated into the signup form as a checkbox for a more streamlined experience.

Note on the note: we'd still need a separate inviter for non members to join Slack, so this is quite low priority.

JamesDearlove committed:
Add code of conduct checkbox

JamesDearlove committed:
Added some padding in between elements

JamesDearlove committed:
Moving to invisible reCAPTCHA

JamesDearlove commented:
![image](https://user-images.githubusercontent.com/39483549/106348542-c45e2f00-6312-11eb-801a-9197a36c6e22.png)

I've addressed most of the concerns. Added a checkbox for the code of conduct, and implemented an invisible reCAPTCHA. The reCAPTCHA will launch a challenge if it deems necessary, it will require updating the reCAPTCHA keys on the server.

In regards to integrating with the signup form, it is pretty straight forward to implement with a Slack API request.

nicklambourne commented on views/index.pug:
```diff
@@ -11,32 +11,49 @@ html
+                  label.checkbox
+                    input(type="checkbox", required)
+                    span#checkboxlabel I have read and understood the 
+                      a(href="https://uqcs.org/code-of-conduct" target="_blank") Code of Conduct.
```
nit:  the full stop should not be part of the link. Try:
```
a(href="https://uqcs.org/code-of-conduct" target="_blank") Code of Conduct
| .
```

nicklambourne commented on views/index.pug:
```diff
@@ -11,32 +11,49 @@ html
+              
+              .buttons.wide.mb-2
+                input(class="button is-link" type="submit", value="Join").submit
+            p(class="help") Already a slack user? 
```
slack -> Slack

nicklambourne commented on views/index.pug:
```diff
@@ -11,32 +11,49 @@ html
+              .buttons.wide.mb-2
+                input(class="button is-link" type="submit", value="Join").submit
+            p(class="help") Already a slack user? 
               a(href=`https://${slackUrl}`) Sign in to #{community} here
```
Full stop after here (but not part of the link; see example in above comment).

nicklambourne reviewed (changes requested):
Just a couple of small nits

JamesDearlove committed:
Fixing up some nitpicks

katrinafyi reviewed (approved):
LGTM. New reCaptcha flow looks much better. Minor nits which might not need to be changed (up to you).

katrinafyi reviewed (commented):
Comment.

JamesDearlove committed:
Removed help class from existing user text
